flocker/provision/_libcloud.py

```diff
@@ -121,3 +123,122 @@ def get_image(driver, image_name):
         return [s for s in driver.list_images() if s.name == image_name][0]
     except IndexError:
         raise ValueError("Unknown image.", image_name)
+
+
+# _node isn't immutable, since libcloud provides new instances
+# with updated data.
+@attributes([
+    Attribute('_node'),
+    Attribute('_provisioner'),
+    'address',
+    'distribution',
+])
+class LibcloudNode(object):
+    """
+    A node created with libcloud.
+
+    :ivar Node _node: The libcloud node object.
+    :ivar LibcloudProvisioner _provisioner: The provisioner that created this
+        node.
+    :ivar bytes address: The IP address of the node.
+    :ivar str distribution: The distribution installed on the node.
+    :ivar bytes name: The name of the node.
+    """
+
+    def destroy(self):
+        """
+        Destroy the node.
+        """
+        self._node.destroy()
+
+    def reboot(self):
+        """
+        Reboot the node.
+        """
+        self._node.reboot()
+
+        self._node, self.addresses = \
+            self._node.driver.wait_until_running([self._node])[0]
+
+    def provision(self, package_source):
+        """
+        Provision flocker on this node.
+        """
+        self._provisioner.provision(
+            node=self,
+            package_source=package_source,
+            distribution=self.distribution,
+        )
+        return self.address
+
+    @property
+    def name(self):
```

Comment:
So node name becomes readonly / immutable. Is that what you meant in the comment above? Maybe add a docstring explaining that.